models/user.go

```diff
@@ -275,24 +275,77 @@ func (u *User) ConfirmEmailChange(tx *storage.Connection, status int) error {
 	u.EmailChangeTokenCurrent = ""
 	u.EmailChangeTokenNew = ""
 	u.EmailChangeConfirmStatus = status
-	return tx.UpdateOnly(
+
+	if err := tx.UpdateOnly(
 		u,
 		"email",
 		"email_change",
 		"email_change_token_current",
 		"email_change_token_new",
 		"email_change_confirm_status",
-	)
+	); err != nil {
+		return err
+	}
+
+	identity, err := FindIdentityByIdAndProvider(tx, u.ID.String(), "email")
+	if err != nil {
+		if IsNotFoundError(err) {
+			// no email identity, not an error
+			return nil
+		}
+
+		return err
+	}
+
+	if _, ok := identity.IdentityData["email"]; ok {
+		identity.IdentityData["email"] = u.EmailChange
```

Comment:
you need to use `u.Email` instead here because `u.EmailChange` has already been [cleared above](https://github.com/supabase/gotrue/pull/824/files#diff-f9c27c5d201f4eb9a23b855d52131aab43a1842aeedcf9356797c4ae75c224dfR273-R274)
```suggestion
		identity.IdentityData["email"] = u.Email
```

Reply:
Done, added tests too. Should have done this from the start.